docs/topics/media-pipeline.rst

```diff
@@ -475,6 +499,30 @@ See here the methods that you can override in your custom Images Pipeline:
     The :class:`ImagesPipeline` is an extension of the :class:`FilesPipeline`,
     customizing the field names and adding custom behavior for images.
 
+   .. method:: file_path(request, response, info)
+
+      This method is called once per downloaded item. It returns the
+      download path of the file originating from the specified
+      :class:`response <scrapy.http.Response>`.
+
+      In addition to ``response``, this method receives the original
+      :class:`request <scrapy.Request>` and
+      :class:`info <scrapy.pipelines.media.MediaPipeline.SpiderInfo>`.
+
+      You can override this method to customize the download path of each file.
+
+      For example, to download all files into the ``files`` folder with their
+      original filenames::
+
+        import os
+        from urllib.parse import urlparse
+
+         def file_path(self, request, response, info):
```

Comment:
Hi. I found this definition wouldn't work, without assigning default values to the keyword arguments ``response`` and ``info``:
```suggestion
         def file_path(self, request, response=None, info=None):
```
Just like in:
https://github.com/scrapy/scrapy/blob/b364bfb68b481a696141aa6b7b3312679670c38e/scrapy/pipelines/files.py#L460

Reply:
I’ll look into it in detail later, but if I recall right those two fields are only needed when using some deprecated methods.

Reply:
Verified, that is only needed to support subclasses of these pipelines that still use the deprecated `file_key` method.